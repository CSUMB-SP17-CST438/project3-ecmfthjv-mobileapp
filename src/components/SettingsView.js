import React, { Component } from 'react';
import { Container, Content, ListItem, Text, Separator, Button } from 'native-base';

export default class SettingsView extends Component {
    constructor(props){
        super(props);
        this.state = {latitude: 'unknown', longitude: 'unknown', lastPosition: 'unknown'}
        //Binds for onclick
        this.testGPSButtonPress = this.testGPSButtonPress.bind(this);
    }
    watchID: ?number = null;

    //Function that sets states with GPS coordinates and other information
    testGPSButtonPress(event){
        navigator.geolocation.getCurrentPosition(
            (position) => {
                //Turns json to string for setting initialPosition state
                this.setState({
                    latitude: position['coords']['latitude'],
                    longitude: position['coords']['longitude']
                });
                var postDetails = {
                    "name": "testPost",
                    "longitude": this.state.longitude,
                    "latitude": this.state.latitude,
                    "content": ":^)"
                };
                var formBody = [];
                for (var property in postDetails) {
                    var encodedKey = encodeURIComponent(property);
                    var encodedValue = encodeURIComponent(postDetails[property]);
                    formBody.push(encodedKey + "=" + encodedValue);
                }
                formBody = formBody.join("&");
                fetch('https://terrasite.herokuapp.com/api/arposts', {
                    method: 'POST',
                    headers: {
                        'Accept': 'application/json',
                        'Content-Type': 'application/x-www-form-urlencoded',
                    },
                    body: formBody
                });
            },
            (error) => alert(JSON.stringify(error)),
            {enableHighAccuracy: true, timeout: 20000, maximumAge: 1000}
        );
        this.watchID = navigator.geolocation.watchPosition((position) => {
            //Turns json to string for setting lastPosition state
            var lastPosition = JSON.stringify(position);
            this.setState({lastPosition});
        });
    }
    render() {
        return (
            <Container>
                <Content>
                    <Separator bordered>
                        <Text>Profile</Text>
                    </Separator>

                    <ListItem >
                        <Text>Sign Out</Text>
                    </ListItem>
                    <ListItem>
                        <Text>Account</Text>
                    </ListItem>
                    <ListItem last>
                        <Text>Display Name</Text>
                    </ListItem>

                    <Separator bordered>
                        <Text>Testing</Text>
                    </Separator>
                    <ListItem>
                        <Button onPress={this.testGPSButtonPress}>
                            <Text>GPS Test</Text>
                        </Button>
                    </ListItem>
                    <ListItem>
                        <Text>Latitude: </Text>
                        <Text>{this.state.latitude}</Text>
                        <Text>Longitude: </Text>
                        <Text>{this.state.longitude}</Text>
                    </ListItem>
                </Content>
            </Container>
        );
    }
}
