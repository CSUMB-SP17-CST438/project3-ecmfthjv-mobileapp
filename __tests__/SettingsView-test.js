import 'react-native';
import React from 'react';
import SettingsView from '../src/components/SettingsView';

import renderer from 'react-test-renderer';

test('SettingsView view renders correctly', () => {
  const tree = renderer.create(
    <SettingsView />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
