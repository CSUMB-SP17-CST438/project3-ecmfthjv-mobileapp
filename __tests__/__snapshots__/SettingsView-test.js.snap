// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsView view renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View>
      <View
        bordered={true}
        style={
          Object {
            "backgroundColor": "#F0EFF5",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "borderTopWidth": 0.5,
            "flex": 1,
            "height": 35,
            "justifyContent": "center",
            "paddingBottom": 11,
            "paddingLeft": 15,
            "paddingTop": 11,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 14,
            }
          }
          virtual={undefined}
        >
          Profile
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 15,
              "paddingBottom": 13,
              "paddingRight": 15,
              "paddingTop": 13,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            virtual={undefined}
          >
            Sign Out
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 15,
              "paddingBottom": 13,
              "paddingRight": 15,
              "paddingTop": 13,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            virtual={undefined}
          >
            Account
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          last={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": -15,
              "paddingBottom": 13,
              "paddingLeft": 30,
              "paddingRight": 15,
              "paddingTop": 13,
              "top": 1,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            virtual={undefined}
          >
            Display Name
          </Text>
        </View>
      </View>
      <View
        bordered={true}
        style={
          Object {
            "backgroundColor": "#F0EFF5",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "borderTopWidth": 0.5,
            "flex": 1,
            "height": 35,
            "justifyContent": "center",
            "paddingBottom": 11,
            "paddingLeft": 15,
            "paddingTop": 11,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 14,
            }
          }
          virtual={undefined}
        >
          Testing
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 15,
              "paddingBottom": 13,
              "paddingRight": 15,
              "paddingTop": 13,
            }
          }
          virtual={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "#007aff",
                "borderColor": "#007aff",
                "borderRadius": 5,
                "borderWidth": null,
                "elevation": 2,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingBottom": 6,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 6,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "lineHeight": 19,
                  "marginLeft": 0,
                  "marginRight": 0,
                }
              }
              virtual={undefined}
            >
              GPS Test
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 15,
              "paddingBottom": 13,
              "paddingRight": 15,
              "paddingTop": 13,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            virtual={undefined}
          >
            Current Position
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            virtual={undefined}
          >
            unknown
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 15,
              "paddingBottom": 13,
              "paddingRight": 15,
              "paddingTop": 13,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            virtual={undefined}
          >
            Last Position
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            virtual={undefined}
          >
            unknown
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 20,
        }
      }
    >
      <View
        danger={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.8)",
            "borderRadius": 5,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 50,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
        success={false}
        virtual={undefined}
        warning={false}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
      </View>
    </View>
  </Modal>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "elevation": 4,
            "height": NaN,
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#757575",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
        <RCTScrollView
          dataArray={Array []}
          dataSource={
            ListViewDataSource {
              "items": 0,
            }
          }
          enableEmptySections={true}
          renderRow={[Function]}
          renderScrollComponent={[Function]}
          style={
            Object {
              "marginLeft": -15,
              "marginRight": -15,
              "marginTop": 15,
            }
          }
          virtual={undefined}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;
